kolibri/core/content/api.py

```diff
@@ -1321,6 +1326,82 @@ def consolidate(self, items, queryset):
         return sorted_items
 
 
+class ContentRequestViewset(ReadOnlyValuesViewset, CreateModelMixin):
+    serializer_class = serializers.ContentDownloadRequestSeralizer
+
+    pagination_class = OptionalPageNumberPagination
+
+    values = (
+        "id",
+        "requested_at",
+        "reason",
+        "contentnode_id",
+        "metadata",
+        "status",
+        "facility",
+        "source_id",
+    )
+
+    def get_queryset(self):
+        return ContentDownloadRequest.objects.filter(source_id=self.request.user.id)
+
+    def annotate_queryset(self, queryset):
+        return (
+            queryset.annotate(
+                has_removal=Exists(
+                    ContentRemovalRequest.objects.filter(
+                        source_model=OuterRef("source_model"),
+                        source_id=OuterRef("source_id"),
+                        contentnode_id=OuterRef("contentnode_id"),
+                        requested_at__gte=OuterRef("requested_at"),
+                    )
+                )
+            )
+            .filter(has_removal=False)
+            .exclude(status=ContentRequestStatus.Failed)
```

Comment:
I think this exclude needs to be inside the subquery, as the failure that @bjester was referring to was the failure of the removal, not the failure of the download.